### PR TITLE
PropertyFilterBuilder: Focus property selector after adding rule or rule group

### PR DESCRIPTION
**Pull request by simasjar**
## Changes
Now property selector in the filter dialog will be focused after adding the rule or rule group.
 

## Testing
Added one test to check if the property selector is focused after adding a new rule.

### CODE REVIEW

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 1]
```diff
@@ -248,4 +249,22 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
+      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+      { actions }
+    );
+
+    const addRuleButton = getByTestId("rule-group-add-rule");
+    addRuleButton.click();
+
+    await waitFor(
+      () =>
+        expect(
```
Comment (saskliutas) [inline ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx:265]: I would suggest checking this with `expect(container.querySelector(".rule-property input") === container.ownerDocument.activeElement).to.be.true;`

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 2]
```diff
@@ -248,4 +249,22 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
   });
+
+  it("focuses rule property after adding new rule", async () => {
+    const actions = new PropertyFilterBuilderActions(sinon.spy());
+    const { getByTestId, container } = renderWithContext(
+      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+      { actions }
+    );
+
+    const addRuleButton = getByTestId("rule-group-add-rule");
```
[the remark this answers is not in the document]
Comment (simasjar) [reply ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx]: fixed

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx (hunk 1 above)
Comment (simasjar) [reply ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx]: fixed

**Review by saskliutas — APPROVED** [review]

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx [hunk 3]
```diff
@@ -206,3 +209,28 @@ const PropertyFilterBuilderGroupOrRule = React.memo(
     );
   }
 );
+
+const useRulePropertyFocus = (
+  currentGroupItemsLength: number,
+  groupRef: React.RefObject<HTMLDivElement>
```
Comment (GerardasB) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx:215]: Usually, I'd expect a hook to return a `React.Ref`. Might be fine in this case, since it is not exported.

**Review by GerardasB — APPROVED** [review]

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx (hunk 3 above)
Comment (saskliutas) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx:215]: Now that you mentioned this maybe it makes sense to change `React.RefObject<HTMLDivElement>` into `HTMLDivElement | null`?

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx [hunk 4]
```diff
@@ -206,3 +212,27 @@ const PropertyFilterBuilderGroupOrRule = React.memo(
...
+    ) {
+      const ruleProperties = groupRef.querySelectorAll<HTMLInputElement>(
+        ".rule-property input"
+      );
+      ruleProperties[ruleProperties.length - 1].focus();
+      isNewRuleAdded.current = false;
+    }
+    previousGroupItemsLength.current = currentGroupItemsLength;
+  }, [currentGroupItemsLength, groupRef]);
+  return isNewRuleAdded;
```
Comment (saskliutas) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx:237]: Why did you change from returning callback to ref?

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx (hunk 3 above)
Comment (simasjar) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx]: applied @saskliutas suggestion

Comment (GerardasB) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx]: I think the current solution might lead to some subtle issues where Ref value is changed w/o a re-render (hence useEffect logic is not run until the next render).

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx [hunk 5]
```diff
@@ -206,3 +212,27 @@ const PropertyFilterBuilderGroupOrRule = React.memo(
     );
   }
 );
+
+const useRulePropertyFocus = (
+  currentGroupItemsLength: number,
+  groupElement: HTMLDivElement | null
+) => {
+  const previousGroupItemsLength = React.useRef<number>(0);
```
Comment (GerardasB) [inline ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx:220]:
```suggestion
const useRulePropertyFocus = (
  currentGroupItemsLength: number,
  groupElement: HTMLDivElement | null
): [() => void, React.Ref<HTMLDivElement>] => {
  const ref= React.useRef<HTMLDivElement>(null);
  return [() => {}, ref];
}
```

I was thinking about something like that?

**Review by GerardasB — APPROVED** [review]

File: ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx (hunk 5 above)
Comment (simasjar) [reply ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx]: I assumed that we don't need `groupElement: HTMLDivElement | null` as a second argument, fixed

**Comment by GerardasB** [comment]
@simasjar please add a note to `NextVersion.md`

**Comment by simasjar** [comment]
> @simasjar please add a note to `NextVersion.md`

Added
